### PR TITLE
Feature: add recurring donation options migration step

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

This adds a recurring donation options migration step to convert v2 form meta to v3 form block attributes.

## Affects

The amount block (recurring donation settings) during form migration.

## Visuals

**v2 Form**

Screenshot 2023-08-30 at 3 32 11 PM

Screenshot 2023-08-30 at 3 33 14 PM

**v3 Form Migrated**

Screenshot 2023-08-30 at 3 32 28 PM

Screenshot 2023-08-30 at 3 33 22 PM

## Testing Instructions

1. Create a v2 form w/ give-recurring installed/enabled
2. Modify the recurring options in small increments followed by running `wp givewp form:migrate {formId}`
3. Check the v3 form to make sure the migrated settings make sense
4. Repeat steps 2-3 until all settings have been tested & compared.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205353925025788
  - https://app.asana.com/0/0/1205396809084390

### CODE REVIEW

File: src/FormMigration/ServiceProvider.php [hunk 1]
```diff
@@ -31,14 +31,16 @@ public function register()
...
                 Steps\FormFields::class,
                 Steps\FormFields\LoginRegistration::class,
                 Steps\FormFields\CompanyDonations::class,
                 Steps\DonationGoal::class,
                 Steps\TermsAndConditions::class,
                 Steps\FormGrid::class,
                 Steps\OfflineDonations::class,
-                Steps\EmailSettings::class,
+                // TODO: this step was causing errors, so I commented it out
+                //Steps\EmailSettings::class,
```
Comment (jonwaldstein) [inline src/FormMigration/ServiceProvider.php:43]: This step was preventing the (migrated) form builder from loading - I think it has something to do with the email settings content not being sanitized for json correctly

**Comment by jonwaldstein** [comment]
@kjohnson this is ready for review.

**Review by kjohnson — COMMENTED** [review]
I would suggest adding methods to the `FormMetaDecorator` instead of using the `getMetaV2` helper method. With the decorator you can type the return and provide docblock details for each decorator method, as opposed to the inline comments for each variable.

**Comment by jonwaldstein** [comment]
> I would suggest adding methods to the `FormMetaDecorator` instead of using the `getMetaV2` helper method. With the decorator you can type the return and provide docblock details for each decorator method, as opposed to the inline comments for each variable.

Good call, updated here 05542818

**Review by kjohnson — APPROVED** [review]
Looks good to me - to QA!

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**